### PR TITLE
Convert to .NET 5

### PR DESCRIPTION
**Pull request by cartermp**
fixes https://github.com/dotnet/try-convert/issues/309

I had to refactor how we deal with TFMs. Because of the lovely TFM change in .NET 5, we now need more information to determine an appropriate TFM. I decided to carry that information around in the `UnconfiguredProject` once it was computed to avoid duplicating the same logic. The upside is that TFM-ery is a bit more centralized now.

- [x] Manual validation
- [x] Update netcoreapp3.1 regression tests
- [x] Add .NET 5 regression tests

### CODE REVIEW

File: src/MSBuild.Abstractions/BaselineProject.cs [hunk 1]
```diff
@@ -12,27 +12,46 @@ public struct BaselineProject
...
+                : AdjustTargetTFM(projectStyle, outputType, candidateTargetTFM);
         }
 
-        public string GetTfm()
+        private static string AdjustTargetTFM(ProjectStyle projectStyle, ProjectOutputType outputType, string candidateTargetTFM)
         {
-            if (GlobalProperties.Contains(MSBuildFacts.TargetFrameworkNodeName, StringComparer.OrdinalIgnoreCase))
+            if (candidateTargetTFM.ContainsIgnoreCase(MSBuildFacts.Net5) && projectStyle == ProjectStyle.WindowsDesktop)
+            {
+                return MSBuildFacts.Net5Windows;
```
Comment (cartermp) [inline src/MSBuild.Abstractions/BaselineProject.cs:32]: I guess I should "future proof" this by appending `"-windows"` to the string?

**Comment by cartermp** [comment]
alright @jmarolf this is done, please observe my crimes

File: global.json [hunk 2]
```diff
@@ -16,4 +16,4 @@
     "Microsoft.DotNet.Arcade.Sdk": "5.0.0-beta.20506.7",
     "Microsoft.DotNet.Helix.Sdk": "5.0.0-beta.20506.7"
   }
-}
```
Comment (jmarolf) [inline global.json:19]: add back that newline

File: src/MSBuild.Abstractions/MSBuild.Abstractions.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (jmarolf) [inline src/MSBuild.Abstractions/MSBuild.Abstractions.csproj:4]: This prevents us from running on the 3.1 runtime. I don't see a reason why someone should require the 5.0 runtime to do conversions. Can't we keep this as 3.1?

File: src/try-convert/try-convert.csproj [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (jmarolf) [inline src/try-convert/try-convert.csproj:5]: As per other comment. can we keep this 3.1?

**Review by jmarolf — CHANGES_REQUESTED** [review]
I don't think this is the approach we want to take. You can upgrade projects to .NET 5 without requiring the .NET 5 runtime. I think we want to add some logic to determine the maximum TFM we can target with the installed SDK and then only offer that.

**Comment by cartermp** [comment]
Is the 5.0 runtime actually required if something is a global tool?

**Comment by cartermp** [comment]
> I think we want to add some logic to determine the maximum TFM we can target with the installed SDK and then only offer that

This is already done. The only time it will go to 5.0 right now is if you have a preview SDK installed and specify the `preview` flag.

File: src/MSBuild.Abstractions/BaselineProject.cs (hunk 1 above)
Comment (cartermp) [reply src/MSBuild.Abstractions/BaselineProject.cs:32]: nah

**Review by jmarolf — APPROVED** [review]

**Pull request merged**